content/asp.net/getting-started/data.md

```diff
@@ -1,20 +1,168 @@
+
+You could just as easily store a connection string in your [application configuration](configuration.md) and connect your context to a SQL database. Rather that using the `Microsoft.EntityFrameworkCore.InMemory` package above we'd use `Microsoft.EntityFrameworkCore.SqlServer` instead, and our call in `ConfigureServices` would look similar to the following:
+
+```
+services.AddDbContext<ErrorContext>(options =>
+    options.UseSqlServer(Configuration.GetConnectionString("ErrorDatabaseConnectionString")));
+```
+
+> Note that the `InMemory` provider is intended for testing and prototyping and is not a production data store. Each time your application starts a new copy of the database is created and, likewise, any data built up at runtime is lost when your process ends.
+
+## Collecting Some Data
+
+The next step is to start adding some data to our database, and we're going to do this through [Middleware](middleware-basic.md).  Create a folder in your project called `Middleware` and then add a new middleware class by right-clicking on the folder and selecting "Add -> New Item". Choose the Middleware class template, and name the class `ErrorCollectingMiddleware`. 
+
+Modify the constructor so that it accepts the `ErrorContext` that we've created and configured:
+
+```c#
+public ErrorCollectingMiddleware(RequestDelegate next, ErrorContext context)
+{
+    _next = next;
+    _context = context;
+}
+```
+
+Next, create a backing field for the context so that we can access it throughout our class:
+
+`private readonly ErrorContext _context;`
+
+Finally, update the `Invoke` method to the following:
+
+```c#
+public async Task Invoke(HttpContext httpContext)
+{
+    await _next(httpContext);
+
+    if (httpContext.Response.StatusCode >= 400)
+    {
+        _context.Errors.Add(new ErrorInformation
+        {
+            Path = httpContext.Request.Path,
+            StatusCode = httpContext.Response.StatusCode
+        });
+        await _context.SaveChangesAsync();
+    }
+}
+```
+
+We won't worry too much about the mechanics here for brevity, but there are two important things to note about our data and our context after we wait for the call `_next` to complete. First, whenever we see an HTTP status code at or above 400 we create a new `ErrorInformation` object on the fly and add it to the `DbSet<ErrorInformation>` property that we have in our context. Secondly, we make a call to `SaveChangesAsync` to commit our rows to the table.
+
+We have to wire in our middleware, which we've [seen before](middleware-basic.md). We want to wrap our execution around the MVC pipeline, so we can add it to the `Configure` method in our `Startup` class, just before the call to `app.UseMvc()`. We can use the extension method that was created for us in our templated middleware class:
 
-Give the reader some additional exercises/tasks they can perform to try out what they've just learned.
\ No newline at end of file
+`app.UseErrorCollectingMiddleware();`
```

Comment:
It's probably worth showing the full 'Configure' method here so they know where to add the code.